org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java

```diff
@@ -0,0 +1,441 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Red Hat Inc. and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
```

Comment:
Please retain the original copyright header, and a reference to the source file this has been copied from. (looks a lot like org.eclipse.jdt.ui/org.eclipse.jdt.internal.ui.text.correction.proposals.NewCUUsingWizardProposal.java

Reply:
`NewCUUsingWizardProposal` is coupled deeply with UI (as implied by its name), so I retain the main stucture and remove the UI related members and logics. The "create change"  part has been totally different from the original one. Do I need to add something in the header, besides reference? Based on the [original header](https://github.com/eclipse/eclipse.jdt.ui/blob/699dba31573baf10ec17af7fabe7165d1e3840ae/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/text/correction/proposals/NewCUUsingWizardProposal.java#L1-L14), it might look like below according to your comments:
```
/*******************************************************************************
 * Copyright (c) 2000, 2018 IBM Corporation and others.  <<<----Do I need to update the year to 2019?
 *
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Originally copied from org.eclipse.jdt.internal.ui.text.correction.proposals.NewCUUsingWizardProposal 

 * Contributors:
 *     Renaud Waldura &lt;renaud+eclipse@waldura.com&gt;
 *     IBM Corporation - updates
 *    <<<--- Do I need to add something here to indicate it's already different from the original one?
 *******************************************************************************/
```

Reply:
Yes add 'Microsoft - reasons for the fork'

Reply:
updated